developer-docs/README.md

```diff
@@ -64,102 +64,162 @@ SEMATIC_API_ADDRESS: https://<my_prod1_host:port>
-of the `github.com/sematic-ai/sematic` repo directory.
+**Note:** Actually pushing the released wheel can only be done if you have
+access to the PyPI repo, which is limited to employees of Sematic.
+
+We cut releases from the `main` branch, following the steps below. They must be
+performed in order, without skipping any. If any errors occur, you must fix
+them and then go back and redo all the steps that have been affected by the
+changes.
+
+1. Bump the version in:
+   - `wheel_constants.bzl` - `wheel_version_string`
+   - `helm/sematic-server/Chart.yaml` - `appVersion`
+   - `sematic/versions.py` - `CURRENT_VERSION`; bump
+     `MIN_CLIENT_SERVER_SUPPORTS` if there are any TODOs mentioning this should
+     be done
+   - `./README.md` - PyPI badge
+
+1. Increment the minor version of the `version` field in
+  `helm/sematic-server/Chart.yaml`. Note that this will be DIFFERENT
+  from the Sematic version you changed in prior steps (chart version
+  will be `1.X.Y`, Sematic version is `0.M.N`).
 
-```bash
-HELM_REPO=~/code/helm-charts
-helm package helm/sematic-server
-helm repo index . --url https://sematic-ai.github.io/helm-charts/sematic-server \
-         --merge $HELM_REPO/index.yaml
-mv index.yaml $HELM_REPO/index.yaml
-mv *.tgz $HELM_REPO/sematic-server/
-```
+1. Update `changelog.md` with the new version number and any missing change
+  entries
 
-You should now have a new `sematic-server/sematic-server-X.X.X.tgz` in the
-`helm-charts` repo, and a modified `index.yaml`.  Commit and push both of
-these to a new branch, and create a PR for the change based on `gh-pages`
-necessary.
-
-Finally, draft the release on GitHub:
-- Add a "What's Changed" section.
-- Add a warning that the client version will need to be upgraded, if this applies.
-- Add a "New Contributors" section, if this applies.
-- Add a "Full Changelog" link.
-- Attach the wheel in the "Assets" section.
+1. Make the release PR, containing the previous changes. After implementing
+  comments and getting approval on the release PR, merge it and pull from the
+  updated `main`. It is mandatory to include this version of main in the
+  subsequent steps. If validation issues are discovered, they must be fixed
+  in patch PRs, and the process restarted from this step.
+
+1. Build the UI:
+    ```bash
+    $ make ui
+    ```
+
+1. Build the wheel:
+    ```bash
+    $ make wheel
+    ```
+
+1. Copy the wheel from `bazel-bin/sematic/sematic-*.whl` into a scratch
+  directory, and use a virtual env to test:
+
+    ```bash
+    $ pip3 install <wheel path>
+    $ sematic stop
+    $ sematic start
+    $ sematic run examples/mnist/pytorch
+    $ sematic stop
+    ```
+
+    Do this for all supported versions of Python. You can check your virtual env
+    Python version using `sematic version` (as well as the Server and Client
+    version).
+
+1. At this point, you should also deploy the Docker image to a cloud Dev environment
```

Comment:
i would also two separate steps here to make sure that we:
1) do a fresh helm install
2) do a helm upgrade from the previous version
both should work.